.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,45 @@
+name: Pull Request
+
+on:
+  pull_request:
+    branches: [ "main" ]
+
+concurrency:
+  group: build-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  detekt:
+    name: Detekt review
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.1.0
+        with:
+          fetch-depth: 1
+          ref: ${{ github.head_ref }}
+
+      - name: Detekt PR Check
```

Comment:
@PSchmiedmayer This is the detekt action that uses [reviewdog](https://github.com/marketplace/actions/run-detekt-with-reviewdog)

Regardless of the linter result, `Detekt review` job will always succeed even if it finds any issues.  There are two options to configure for `reviewdog_reporter: (github-pr-check, github-pr-review)` and I am using `github-pr-check` github-pr-check reporter reports results to [GitHub Checks](https://github.com/reviewdog/reviewdog#reporter-github-checks--reportergithub-pr-check).

Detekt gradle task is also included in the `build` task of the project, hence if there are any issues `Build and test` job fails as expected due to detekt issues. Ideally, we shouldn't proceed with `Build and test` if we already seen issues in the `Detekt review` job. I tried multiple configurations but nothing worked out so far, left both jobs to run in parallel for now. I have intentionally included several warnings in `CoroutinesModule.kt` file in this PR for testing purposes.

Reply:
Please have a look at this pipeline: https://github.com/StanfordSpezi/SpeziKt/actions/runs/9309515966/job/25625090503

Even though we define just two jobs, the detekt action running in `Detekt review` job, additionally creates a third `detekt` job. This is the job that fails actually if there are some violations. Ideally here we should stop the whole pipeline if that's the case, however I did not succeed to do that, even when specifying a fail fast strategy.

> I am wondering if we could simply run `./gradlew detekt` right after this action as part of this job as a separate run step after this one?

That is not necessary in my opinion. The detekt github action is way more lightweight in comparison to  `./gradlew detekt` gradle task. The github action operates on the source code directly rather than on artifacts and does not require a project build. It executes the packaged JAR java application: `java -jar /opt/detekt.jar --config internal/detekt-config.yml --report xml:detekt_report.xml --excludes **/build/**,**/.idea/** --plugins /opt/detekt-formatting.jar`.

On the other hand, `./gradlew detekt` would require a project build beforehand in order to run. We could theoretically do that, but it would be redundant in my opinion as that detekt gradle task is already part of the `build` (`check`) gradle task (See [here](https://detekt.dev/docs/gettingstarted/gradle/#disabling-detekt-from-the-check-task)). We can exclude as proposed in the link and we would need to explicitly run `./gradlew detekt` after `build`

I think if everyone will follow the guide with setting up detekt locally (Android studio plugin and the git hook), it will ideally never be the case that the detekt job will fail.